pkg/sdn/plugin/subnets.go

```diff
@@ -205,6 +206,47 @@ func (node *OsdnNode) initSelfSubnet() error {
 	return nil
 }
 
+// Only run on the master
+// Watch for all hostsubnet events and if one is found with the right annotation, use the IPAM to dole a real subnet
+func (master *OsdnMaster) watchSubnetsOnMaster() {
+	eventQueue := master.registry.RunEventQueue(HostSubnets)
+
+	for {
+		eventType, obj, err := eventQueue.Pop()
+		if err != nil {
+			utilruntime.HandleError(fmt.Errorf("EventQueue failed for subnets: %v", err))
+			return
+		}
+		hs := obj.(*osapi.HostSubnet)
+		name := hs.ObjectMeta.Name
+		hostIP := hs.HostIP
+
+		log.V(5).Infof("Watch %s event for HostSubnet %q", strings.Title(string(eventType)), name)
+		switch eventType {
+		case watch.Added, watch.Modified:
+			if err = master.registry.ValidateNodeIP(hostIP); err != nil {
+				log.Errorf("Ignoring invalid subnet for node %s: %v", hostIP, err)
+				continue
+			}
+
+			if _, ok := hs.Annotations["hostsubnet.sdn.openshift.io/autocreate"]; ok {
```

Comment:
Yeah, needs a constant, and we should be consistent with other constants like the macvlan one.